tests/vault.test.ts

```diff
@@ -222,6 +237,67 @@ describe("Vault", () => {
 				"Assets should be the amount of assets requested"
 			).to.equal(ethers.utils.parseUnits("500", underlyingDecimals));
 		});
+
+		it("Should not allow user to withdraw before the lock up period ends", async () => {
+			const amount = ethers.utils.parseUnits("1000", underlyingDecimals);
+			await underlying.connect(alice).approve(vault.address, amount);
+
+			const latestBlockNumber = await ethers.provider.getBlockNumber();
+			const latestBlock = await ethers.provider.getBlock(latestBlockNumber);
+			await vault.connect(alice).deposit(amount, alice.address);
+
+			const lockedTime = await (
+				await vault.lockedTime(alice.address)
+			).toNumber();
```

Comment:
the first await doesn't do anything here. For shorter syntax, you could do `+(await vault.lockedTime(alice.address)`